src/constants/triggers.ts

```diff
@@ -0,0 +1,218 @@
+
+export const buildRegexSelectors: RegexSelector[] = [
+  {
+    type: "display-name",
+    typeLabel: "Build Variant Name",
+  },
+  {
+    type: "build-variant",
+    typeLabel: "Build Variant ID",
+  },
+];
+
+export const taskRegexSelectors: RegexSelector[] = [
+  {
+    type: "display-name",
+    typeLabel: "Task Name",
+  },
+  {
+    type: "build-variant",
+    typeLabel: "Build Variant Name",
+  },
+];
+
+export const failureTypeSubscriberConfig = {
+  text: "Failure Type",
+  key: "failure-type",
+  type: "select",
+  options: {
+    any: "Any",
+    test: "Test",
+    system: "System",
+    setup: "Setup",
+  },
+  default: "any",
+};
+
+export const requesterSubscriberConfig = {
+  text: "Build Initiator",
+  key: "requester",
+  type: "select",
+  options: {
+    gitter_request: "Commit",
+    patch_request: "Patch",
+    github_pull_request: "Pull Request",
+    merge_test: "Commit Queue",
+    ad_hoc: "Periodic Build",
+  },
+  default: "gitter_request",
+};
+
+export const clearExtraFieldsInputCb = (accum: StringMap, eF: ExtraField) => ({
+  ...accum,
+  [eF.key]: "10",
+});
+
+export const projectTriggers: Trigger[] = [
+  {
+    trigger: TriggerType.OUTCOME,
+    resourceType: ResourceType.VERSION,
+    label: "Any Version Finishes",
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.FAILURE,
+    resourceType: ResourceType.VERSION,
+    label: "Any Version Fails",
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.OUTCOME,
+    resourceType: ResourceType.BUILD,
+    label: "Any Build Finishes",
+    regexSelectors: buildRegexSelectors,
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.FAILURE,
+    resourceType: ResourceType.BUILD,
+    label: "Any Build Fails",
+    regexSelectors: buildRegexSelectors,
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.OUTCOME,
+    resourceType: ResourceType.TASK,
+    label: "Any task finishes",
+    regexSelectors: taskRegexSelectors,
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.FAILURE,
+    resourceType: ResourceType.TASK,
+    label: "Any Task Fails",
+    regexSelectors: taskRegexSelectors,
+    extraFields: [failureTypeSubscriberConfig, requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.FIRST_FAILURE_BUILD,
+    resourceType: ResourceType.TASK,
+    label: "The First Failure In a Version Occurs",
+    regexSelectors: taskRegexSelectors,
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.FIRST_FAILURE_BUILD,
+    resourceType: ResourceType.TASK,
+    label: "The First Failure In Each Build Occurs",
+    regexSelectors: taskRegexSelectors,
+    extraFields: [requesterSubscriberConfig],
+  },
+  {
+    trigger: TriggerType.FIRST_FAILURE_VERSION_NAME,
+    resourceType: ResourceType.TASK,
+    label: "The First Failure In Each Version For Each Task Name Occurs",
```

Comment:
tiny nit: shouldn't capitalize "in" or "for" (also for the other labels)
https://www.webucator.com/article/how-to-capitalize-headings-and-titles/